### PR TITLE
Js3week3

### PR DESCRIPTION
**Pull request by priyankapandey13**

### CODE REVIEW

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    this.name = name;
+    this.price = price;
+  }
+
+  convertToCurrency(currency) {
+    fetch(`https://api.exchangerate-api.com/v4/latest/DKK`)
+      .then((response) => response.json())
+      .then((data) => {
+        let currencyConvert = (data.rates[currency] * this.price).toFixed(2);
+        console.log(
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping-cart/main.js]: usually you want to do a return here and console log outside the function (where you're calling it from)

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  }
+
+  addProduct(product) {
+    // Implement functionality here
+    this.products.push(product);
+  }
+
+  removeProduct(product) {
+    // Implement functionality here
+    const removedProduct = this.products.filter(
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping-cart/main.js]: this is confusing. You call the variable `removedProduct`, but it actually contains the _opposite_ of that, i.e. everything _except_ the removed product. A better name would be `remainingProducts` or something like that.

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+      (itemname) => product !== itemname
+    );
+    this.products = removedProduct;
+  }
+
+  searchProduct(productName) {
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping-cart/main.js:41]: this should probably return something.

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    );
+    console.log(SearchProductName);
+  }
+
+  getTotal() {
+    // Implement functionality here
+    const pricelist = this.products.map((product) => product.price);
+    const TotalPrice = pricelist.reduce((sum, item) => sum + item);
+    console.log(TotalPrice);
+    return TotalPrice;
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping-cart/main.js:54]: yes returning the price is the right way to do it.

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    // Implement functionality here
+    let SearchProductName = this.products.filter(
+      (itemname) => itemname === productName
+    );
+    console.log(SearchProductName);
+  }
+
+  getTotal() {
+    // Implement functionality here
+    const pricelist = this.products.map((product) => product.price);
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping-cart/main.js]: good variable name, but it is not necessary to put parentheses around `(product)`

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+shoppingCart.addProduct(Keyboard);
+
+// console.log(shoppingCart);
+
+// shoppingCart.removeProduct('Playstation');
+// shoppingCart.searchProduct('Playstation');
+// shoppingCart.getTotal();
+// shoppingCart.renderProducts();
+
+shoppingCart.getUser();
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping-cart/main.js]: here you probably want to get a return value and then call `renderUser` with that value. So something like
`const userData = shoppingCard.getUser();
renderUser(userData);`

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+    });
+  }
+
+  getUser() {
+    // Implement functionality here
+    fetch(`https://jsonplaceholder.typicode.com/users/1`)
+      .then((Response) => Response.json())
+      // .then(data=> console.log(data))
+      .then((user) => {
+        this.renderuser(user);
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping-cart/main.js]: i would not call `renderUser` from here, but from where you're calling getUser (see my other comment). 
Why? Usually you want to separate data operations (numbers, text), from DOM operations (createElement, appendChild, innerText, etc)

File: javascript/javascript3/week3/shopping-cart/main.js [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+const shoppingCart = new ShoppingCart("Phone", 3000);
+const flatscreen = new Product("flat-screen", 5000);
+// shoppingCart.addProduct(flatscreen);
+
+const Phone = new Product("Phone", 1000);
+const Fridge = new Product("Fridge", 2000);
+const laptop = new Product("laptop", 3500);
+const TV = new Product("TV", 3000);
+const Playstation = new Product("Playstation", 1000);
+const Keyboard = new Product("Keyboard", 800);
```
Comment (aleksd-dfds) [inline javascript/javascript3/week3/shopping-cart/main.js:97]: all good here

**Review by aleksd-dfds — CHANGES_REQUESTED** [review]
OK a lot of good things here, i made a few corrections. Seems like you put the react homework here as well which i'm not going to grade. I recommend you brush up on your Git skills and do your commits in a more clean way. As a grader i should be able to see:
- What are you changes from last time? (not the full files from scratch)
- No other unnecessary files from other weeks or homeworks.

File: javascript/javascript3/week3/shopping-cart/main.js (hunk 2 above)
Comment (priyankapandey13) [reply javascript/javascript3/week3/shopping-cart/main.js]: Done, I renamed it updatedArrayList now

**Pull request merged**